components/feature/push/src/main/java/mozilla/components/feature/push/AutoPushFeature.kt

```diff
@@ -230,6 +230,31 @@ class AutoPushFeature(
         }
     }
 
+    /**
+     * Checks if a subscription for the [scope] already exists.
+     *
+     * @param scope The subscription identifier which usually represents the website's URI.
+     * @param appServerKey An optional key provided by the application server.
+     * @param block The callback invoked when a subscription for the [scope] is found, otherwise null.
+     */
+    fun getSubscription(
+        scope: String,
+        appServerKey: String? = null,
+        block: (AutoPushSubscription?) -> Unit
+    ) {
+        connection.ifInitialized {
```

Comment:
So the `block` may never even get called?

Reply:
Yes, it could if the push native layer is not initialized, but we would have much bigger problems here if we can get this far in the call stack having not crashed yet.

This has come up before so maybe we call simplify this call by putting it within the `Connection`. It was meant to satisfy the push state along and remove the null-ability when needing to use the native API.

Reply:
Filed https://github.com/mozilla-mobile/android-components/issues/7184 to track this work.